bitcoin/src/blockdata/witness.rs

```diff
@@ -323,6 +323,23 @@ impl Witness {
         self.push(&sig[..signature.len() + 1]);
     }
 
+    /// Pushes a segwit v0 signature on the witness, requires an allocation.
+    pub fn push_segwit_v0_signature(&mut self, sig: &SegwitV0Signature) {
+        let signature = sig.0.sig.serialize_der();
+        let hash_type = sig.0.hash_ty;
+
+        // Note that a maximal length ECDSA signature is 72 bytes, plus the sighash type makes 73
+        let mut sig = [0; 73];
+        sig[..signature.len()].copy_from_slice(&signature);
+        sig[signature.len()] = hash_type as u8;
+        self.push(&sig[..signature.len() + 1]);
+    }
+
+    /// Pushes a taproot signature on the witness, requires an allocation.
+    pub fn push_taproot_signature(&mut self, sig: &taproot::Signature) {
```

Comment:
Note to self, this is wrong, its only keys path spend not script path spend so the name is wrong.